src/syscheckd/create_db.c

```diff
@@ -434,9 +461,11 @@ static fim_sanitize_state_t fim_process_file_from_db(const char *path, OSList *s
             return FIM_FILE_ERROR;
         }
 
+        w_rwlock_rdlock(&syscheck.directories_lock);
```

Comment:
The only possible way to `fim_process_file_from_db` is through `fim_checker` which is called by holding `syscheck.directories_lock` for reading, we should remove handling of the mutex here and check no race conditions are possible in favor of simplifying code.